functions/src/chat.ts

```diff
@@ -0,0 +1,47 @@
+import type { DocumentReference } from "firebase-admin/firestore";
+import { db } from "./app";
+import { getChatRoomCollection, getChatRoomMessagesCollection } from "./firestore-collections";
+import { Lobby } from "./firestore-types/lobby";
+
+export async function deleteChatRooms(lobbyDoc: DocumentReference<Lobby>) {
+  const lobbyDocSnapshot = await lobbyDoc.get();
+  return db.runTransaction(async () => {
```

Comment:
you are not actually using the transaction fo example instead of doing
```ts
const messages = await getChatRoomMessagesCollection(room).get();
```
you should do
```ts
const messages = await transaction.get(getChatRoomMessagesCollection(room));
```
you can look at `startPrompt` or `collectPromptAnswers` in `lobby.ts` for an example of how to di it correctly or you can read the documentation here: https://firebase.google.com/docs/firestore/manage-data/transactions